src/input/type_base.hh

```diff
@@ -162,6 +168,14 @@ public:
     /// Print JSON output of attributes to @p stream.
     void write_attributes(ostream& stream) const;
 
+    /// Create instance of generic type
+    virtual boost::shared_ptr<TypeBase> make_instance(std::vector<ParameterPair> vec = std::vector<ParameterPair>()) const
+    { ASSERT(false, "make_instance() may not be called for TypeBase class.\n"); };
+
+    /// Return if type is generic.
+    inline bool is_generic() const
```

Comment:
It seems that this method is not used. We will see if it is necessary somewhere during finish phase.